### PR TITLE
Feature/15 ipc create wrapper class for unix message queue

### LINKED ISSUE #15: IPC - Create wrapper class for unix message queue

### PR DESCRIPTION
**Pull request by Cavonstavant**
Closes #15
Missing unit tests

### CODE REVIEW

File: src/MessageQueue/MessageQueue.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+}
+void MessageQueue::sendMessage(const std::string &message_text, long message_type)
+{
+    int message_id;
+
+    if (message_text.length() >= MAX_MESSAGE_SIZE)
+        throw PlazzaEX("Message is too long", Logger::MEDIUM);
+    message_id = msgget(this->_key, 0666 | IPC_CREAT);
+    if (message_id == -1) {
+        exit(84);
```
Comment (osvegn) [inline src/MessageQueue/MessageQueue.cpp:31]: Throw an error before

File: src/MessageQueue/MessageQueue.cpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    }
+    _msg = new MessageQueue::message{.type = message_type};
+    std::strcpy(_msg->text, message_text.c_str());
+    if (msgsnd(message_id, (void *) &_msg, sizeof(_msg->text), 0) == -1) {
+        exit(84);
```
Comment (osvegn) [inline src/MessageQueue/MessageQueue.cpp:36]: Same

File: src/MessageQueue/MessageQueue.cpp [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    delete _msg;
+}
+
+std::string MessageQueue::receiveMessage(long message_type)
+{
+    int message_id = msgget(this->_key, 0666 | IPC_CREAT);
+    MessageQueue::message *msg = nullptr;
+
+    if (message_id == -1) {
+        exit(84);
```
Comment (osvegn) [inline src/MessageQueue/MessageQueue.cpp:47]: Same

**Review by osvegn — CHANGES_REQUESTED** [review]
Why exit(84) is called, but without throwing an error ?

**Comment by Cavonstavant** [comment]
because

File: tests/unit_tests/MessageQueue/MessageQueue_test.cpp [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+** EPITECH PROJECT, 2022
+** plazza
+** File description:
+** messageQueue_test
+*/
+
+#include <criterion/criterion.h>
+// #include "MessageQueue.hpp"
+
+Test(MessageQueue, test_messageQueue_get_size)
```
Comment (osvegn) [inline tests/unit_tests/MessageQueue/MessageQueue_test.cpp:11]: Need a valide test

**Review by osvegn — CHANGES_REQUESTED** [review]
All is good, but need to do a valid test in message queue

**Review by osvegn — APPROVED** [review]
LGTM

**Pull request merged**